lupa-core/src/main/kotlin/org/jetbrains/research/lupa/kotlinAnalysis/AnalysisExecutor.kt

```diff
@@ -32,23 +34,37 @@ abstract class AnalysisExecutor {
     /** Executes analysis for all projects in [given directory][projectsDir]. */
     fun execute(
         projectsDir: Path,
-        repositoryOpener: (Path, (Project) -> Unit) -> Unit = RepositoryOpenerUtil.Companion::standardRepositoryOpener
+        repositoryOpener: (Path, (Project) -> Unit, ((GitRepository) -> Unit)?) -> Unit =
+            RepositoryOpenerUtil.Companion::standardRepositoryOpener,
     ) {
         init()
         try {
-            repositoryOpener(projectsDir, ::analyse)
+            repositoryOpener(projectsDir, ::analyse) { repo -> configData?.postExecuteAction(repo) }
         } finally {
             close()
         }
     }
 }
 
 /** Class for simultaneous execution of multiple analysis for each project in given dataset. */
-class MultipleAnalysisExecutor(private val analysisExecutors: List<AnalysisExecutor>) : AnalysisExecutor() {
+class MultipleAnalysisExecutor(
+    private val analysisExecutors: List<AnalysisExecutor>,
+    configData: Configurable? = null,
+) : AnalysisExecutor(configData) {
 
     override fun analyse(project: Project) {
         analysisExecutors.forEach { it.analyse(project) }
     }
 
     override val controlledResourceManagers = analysisExecutors.flatMap { it.controlledResourceManagers }.toSet()
 }
+
+data class KotlinTeamConfiguration(val connection: DatabaseConnection? = null) : Configurable {
+    override fun postExecuteAction(repo: GitRepository) {
+        connection?.updateRepoDate(repo)
+    }
+}
+
+interface Configurable {
```

Comment:
What about `IExecutorHelper`? It is more suitable for the `postExecuteAction` function

And 
```
abstract class AnalysisExecutor(protected open val executorHelper: Executable? = null) 
```

And `data class KotlinTeamExecutorHelper`